packages/std/src/mock.rs

```diff
@@ -128,20 +129,26 @@ pub fn mock_env<T: Api, U: Into<HumanAddr>>(api: &T, sender: U, sent: &[Coin]) -
 
 /// MockQuerier holds an immutable table of bank balances
 /// TODO: also allow querying contracts
-#[derive(Clone, Default)]
-pub struct MockQuerier {
+pub struct MockQuerier<C: Clone + DeserializeOwned = Never> {
     bank: BankQuerier,
     staking: StakingQuerier,
     // placeholder to add support later
     wasm: NoWasmQuerier,
+    // TODO: Replace argument C -> &C. This requires proper lifetime handling ...
+    handle_custom: Box<dyn Fn(C) -> QuerierResult>, // use box to avoid the need of another generic type
```

Comment:
@reuvenpo I'd like to change the function argument of the custom query `C` to `&C`. Any idea how to tell Rust:
- the `handle_custom` function itself is owned by the `MockQuerier` instance and lives as long as it (same is current code)
- the argument only needs to live as long as one execution of the function

I cannot get this to work. Rust suggests adding explicit lifetimes to `MockQuerier`, which would drive me crazy if I had to set them everywhere `MockQuerier` is used.

Reply:
The `for` keyword can be used to introduce HRTB (Higher Ranked Trait Bounds):
https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=dc56c379c30286b23ef73b5db99cf39d

Reply:
Perfect, thanks a lot! Implemented in https://github.com/CosmWasm/cosmwasm/pull/357/commits/2d737c3d8d03f5ca6dcc1e50f637c1421ad0acb5

I tried something like this before but then ran into trouble with the unused argument situation (where many open Rust issues exist). Since I did not know what I was doing, I gave up too early.